yt_dlp/extractor/gofile.py

```diff
@@ -0,0 +1,65 @@
+# coding: utf-8
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    try_get
+)
+
+
+class GofileIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?gofile\.io/d/(?P<id>[^/]+)'
+    _TESTS = [{
+        'url': 'https://gofile.io/d/AMZyDw',
+        'info_dict': {
+            'id': 'AMZyDw',
+        },
+        'playlist_mincount': 2,
+        'playlist': [{
+            'info_dict': {
+                'id': 'de571ac1-5edc-42e2-8ec2-bdac83ad4a31',
+                'filesize': 928116,
+                'ext': 'mp4',
+                'title': 'nuuh'
+            }
+        }]
+    }]
+
+    def _entries(self, file_id):
+
+        # Create guest account
+        accountdata = self._download_json('https://api.gofile.io/createAccount', 'Gofile', note='Getting a new guest account')
+        accountToken = accountdata['data']['token']
```

Comment:
move this to `_real_initialize`. That way if user gives `yt-dlp URL1 URL2`, the token is only generated once. Also, if user gives cookies, use that token instead of generating a new one